core/src/settings.rs

```diff
@@ -111,10 +123,25 @@ impl RuntimeSettings {
     pub(crate) fn to_eal_args(&self) -> Vec<String> {
         let mut eal_args = vec![];
 
-        // add the app name
+        // adds the app name
         eal_args.push(self.app_name.clone());
 
-        // add all the ports
+        let proc_type = if self.secondary {
+            "secondary".to_owned()
+        } else {
+            "primary".to_owned()
+        };
+
+        // adds the proc type
+        eal_args.push("--proc-type".to_owned());
+        eal_args.push(proc_type);
+
+        // adds the mem file prefix
+        let prefix = self.app_group.as_ref().unwrap_or(&self.app_name);
```

Comment:
this just makes file-prefix that is exposed only to dpdk as an option, yeah?

Reply:
yeah and a slightly abstraction. not directly exposing as `file-prefix` may let us do other interesting things. like we can maybe tag the metrics a certain way for multi-proc apps for example.